pkg/shell/nav.tsx

```diff
@@ -127,10 +183,10 @@ export class CockpitNav extends React.Component {
     }
 
     render() {
-        const groups = [];
+        const groups: ItemGroup<NavItem>[] = [];
         const term = this.state.search.toLowerCase();
         this.props.groups.forEach(g => {
-            const new_items = g.items.map(i => this.props.filtering(i, term)).filter(Boolean);
+            const new_items = g.items.map(i => this.props.filtering(i, term)).filter((i): i is NavItem => !!i);
```

Comment:
It makes no difference! :-O 

I put it there to tell TS that there are no nulls in the result of filtering anymore so that new_items gets type `NavItem[]`.

But, just rewriting this from `Boolean` to `i => !!i` is enough to clue in TS. No type guard needed.